pypeit/core/wavecal/kdtree_generator.py

```diff
@@ -0,0 +1,297 @@
+"""This script is used to generate the KD Tree that is needed for
+the kdtree pattern matching wavelength calibration algorithm. At
+present, this method is only used for calibrating ThAr lamps.
+
+You should not run this script unless you know what you're doing,
+since you could mess up the ThAr patterns that are used in the
+wavelength calibration routine. This script should not be called
+from within PypeIt - it should be run as a standalone script, and
+it's only purpose is to generate a KD Tree with the desired patterns.
+"""
+
+from pypeit.core.wavecal import waveio
+from astropy.table import vstack
+import numba as nb
+from scipy.spatial import cKDTree
+import numpy as np
+import pickle
+
+
+@nb.jit(nopython=True, cache=True)
+def trigon(linelist, numsrch, maxlin):
+    """
```

Comment:
describe the other 2 parameters too, please

Reply:
OK, done!